### ISSUE TITLE
dashboard: use pre-generated exposure tables

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1088]
Use the $DESI_SPECTRO_REDUX/$SPECPROD/exposure_tables/ to know what exposures should exist for a production instead of scanning the raw data headers every time.

@zkdtc please coordinate with @akremin for how the dashboard should know that an exposure is flagged as bad and thus is ok that it isn't processed (in which case it probably shouldn't appear as a row in the dashboard at all).

**Comment by zkdtc** [comment]
Have switched to using exptable. For two testing night: 20210115 and 20210116, the processing time is reduced from 20s to <1s. However, there is no 'FLAVOR' in the current exptable, they are replaced using 'NULL' for now. Also find inconsistency in SPCGRPHS in exptable and in the header. Expid 72716 for example, SPCGRPHS = 1-9 in the table while =0-9 in the header.

**Comment by akremin** [comment]
I believe the spectrographs difference is intentional, but we should discuss whether we want to change it. If the user specifies the spectrographs, it uses that instead of inferring it from the data. But there is also the "camword" which is what is used for processing. We could have spectrographs summarize what data is there and then camword be the restricted set of user-defined cameras for processing.

For posterity-- the reason there is a difference here is because petal 0 and 3 each had issues on various nights and weren't positioned, but the spectrographs still acquired light (so the petal was effectively 500 random sky fibers). Some nights we just ran that petal anyway, on a few nights we restricted to save on computational time.
